module/planning/PlanKick/src/PlanKick.cpp

```diff
@@ -0,0 +1,105 @@
+            cfg.kick_leg                = config["kick_leg"].as<std::string>();
+        });
+
+        on<Provide<KickTo>, Trigger<FilteredBall>, Needs<Kick>>().then(
+            [this](const KickTo& kick, const RunInfo& info, const FilteredBall& ball) {
+                // If the kicking subtask is done, then we are no longer kicking
+                if (info.run_reason == RunInfo::RunReason::SUBTASK_DONE) {
+                    kicking = false;
+                }
+
+                // If we're already kicking, don't do anything
+                if (kicking) {
+                    emit<Task>(std::make_unique<Idle>());
+                    return;
+                }
+
+                // CHECK IF BALL IS BALL MEASUREMENT IS RECENT ENOUGH
+                // If the ball measurement is old, then don't do anything
+                auto time_difference = std::chrono::duration_cast<std::chrono::milliseconds>(
+                    NUClear::clock::now() - ball.time_of_measurement);
+                if (time_difference.count() >= cfg.timeout_threshold) {
+                    emit<Task>(std::make_unique<Idle>());
+                    return;
+                }
+
+                // CHECK IF CLOSE TO BALL
+                // Get the angle and distance to the ball
+                float ball_angle    = std::abs(std::atan2(ball.rBTt.y(), ball.rBTt.x()));
+                float ball_distance = ball.rBTt.head(2).norm();
+
+                // Need to be near the ball to consider kicking it
+                if (ball_distance > cfg.ball_distance_threshold || ball_angle > cfg.ball_angle_threshold) {
+                    emit<Task>(std::make_unique<Idle>());
+                    return;
+                }
+
+                // CHECK IF FACING POINT TO KICK TO
+                float align_angle = std::abs(std::atan2(kick.rPTt.y(), kick.rPTt.x()));
+
+                // Don't kick if we should align but we're not aligned to the target
+                if (cfg.align && align_angle > cfg.target_angle_threshold) {
+                    emit<Task>(std::make_unique<Idle>());
+                    return;
+                }
+
+                // ALL CHECKS PASSED, KICK!
+                // Set kicking flag so we don't keep trying to kick
+                if (!kicking) {
+                    kicking = true;
+                }
+
+                // Check if config specifies a leg
+                if (cfg.kick_leg == "left") {  // can only kick with left
+                    emit<Task>(std::make_unique<Kick>(LimbID::LEFT_LEG));
+                    return;
+                }
+                else if (cfg.kick_leg == "right") {  // can only kick with right
+                    emit<Task>(std::make_unique<Kick>(LimbID::RIGHT_LEG));
+                    return;
+                }
+
+                // Else determine leg based on ball position
+                // If the ball is more to the left, then kick with the left leg
+                if (ball.rBTt.y() > 0.0) {
```

Comment:
May as well combine these with the ones above and delete those
```suggestion
                if (cfg.kick_leg == "left" || ball.rBTt.y() > 0.0) {
```

Reply:
This doesn't quite work, if it's more to the left but right has been forced in config, it'll kick with left still. We'd wants something like 
 ```suggestion
                if (cfg.kick_leg == "left" || (cfg.kick_leg == "auto" && ball.rBTt.y() > 0.0)) {
```